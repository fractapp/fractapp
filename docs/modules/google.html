<!doctype html>
<html class="default no-js">
<head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<title>Google | fractapp</title>
	<meta name="description" content="Documentation for fractapp">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="../assets/css/main.css">
</head>
<body>
<header>
	<div class="tsd-page-toolbar">
		<div class="container">
			<div class="table-wrap">
				<div class="table-cell" id="tsd-search" data-index="../assets/js/search.json" data-base="..">
					<div class="field">
						<label for="tsd-search-field" class="tsd-widget search no-caption">Search</label>
						<input id="tsd-search-field" type="text" />
					</div>
					<ul class="results">
						<li class="state loading">Preparing search index...</li>
						<li class="state failure">The search index is not available</li>
					</ul>
					<a href="../index.html" class="title">fractapp</a>
				</div>
				<div class="table-cell" id="tsd-widgets">
					<div id="tsd-filter">
						<a href="#" class="tsd-widget options no-caption" data-toggle="options">Options</a>
						<div class="tsd-filter-group">
							<div class="tsd-select" id="tsd-filter-visibility">
								<span class="tsd-select-label">All</span>
								<ul class="tsd-select-list">
									<li data-value="public">Public</li>
									<li data-value="protected">Public/Protected</li>
									<li data-value="private" class="selected">All</li>
								</ul>
							</div>
							<input type="checkbox" id="tsd-filter-inherited" checked />
							<label class="tsd-widget" for="tsd-filter-inherited">Inherited</label>
							<input type="checkbox" id="tsd-filter-externals" checked />
							<label class="tsd-widget" for="tsd-filter-externals">Externals</label>
							<input type="checkbox" id="tsd-filter-only-exported" />
							<label class="tsd-widget" for="tsd-filter-only-exported">Only exported</label>
						</div>
					</div>
					<a href="#" class="tsd-widget menu no-caption" data-toggle="menu">Menu</a>
				</div>
			</div>
		</div>
	</div>
	<div class="tsd-page-title">
		<div class="container">
			<ul class="tsd-breadcrumb">
				<li>
					<a href="../globals.html">Globals</a>
				</li>
				<li>
					<a href="google.html">Google</a>
				</li>
			</ul>
			<h1>Namespace Google</h1>
		</div>
	</div>
</header>
<div class="container container-main">
	<div class="row">
		<div class="col-8 col-content">
			<section class="tsd-panel tsd-comment">
				<div class="tsd-comment tsd-typography">
					<dl class="tsd-comment-tags">
						<dt>namespace</dt>
						<dd></dd>
					</dl>
				</div>
			</section>
			<section class="tsd-panel-group tsd-index-group">
				<h2>Index</h2>
				<section class="tsd-panel tsd-index-panel">
					<div class="tsd-index-content">
						<section class="tsd-index-section ">
							<h3>Variables</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-variable tsd-parent-kind-namespace"><a href="google.html#foldertype" class="tsd-kind-icon">folder<wbr>Type</a></li>
								<li class="tsd-kind-variable tsd-parent-kind-namespace"><a href="google.html#jsontype" class="tsd-kind-icon">json<wbr>Type</a></li>
							</ul>
						</section>
						<section class="tsd-index-section ">
							<h3>Functions</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-function tsd-parent-kind-namespace"><a href="google.html#getfilebackup" class="tsd-kind-icon">get<wbr>File<wbr>Backup</a></li>
								<li class="tsd-kind-function tsd-parent-kind-namespace"><a href="google.html#getitems" class="tsd-kind-icon">get<wbr>Items</a></li>
								<li class="tsd-kind-function tsd-parent-kind-namespace"><a href="google.html#safesave" class="tsd-kind-icon">safe<wbr>Save</a></li>
								<li class="tsd-kind-function tsd-parent-kind-namespace"><a href="google.html#signin" class="tsd-kind-icon">sign<wbr>In</a></li>
								<li class="tsd-kind-function tsd-parent-kind-namespace"><a href="google.html#signout" class="tsd-kind-icon">sign<wbr>Out</a></li>
							</ul>
						</section>
					</div>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Variables</h2>
				<section class="tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-namespace">
					<a name="foldertype" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagConst">Const</span> folder<wbr>Type</h3>
					<div class="tsd-signature tsd-kind-icon">folder<wbr>Type<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">"application/vnd.google-apps.folder"</span><span class="tsd-signature-symbol"> = &quot;application/vnd.google-apps.folder&quot;</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/fractapp/fractapp/blob/77ffacb/src/utils/google.ts#L18">src/utils/google.ts:18</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-namespace">
					<a name="jsontype" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagConst">Const</span> json<wbr>Type</h3>
					<div class="tsd-signature tsd-kind-icon">json<wbr>Type<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">"application/json"</span><span class="tsd-signature-symbol"> = &quot;application/json&quot;</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/fractapp/fractapp/blob/77ffacb/src/utils/google.ts#L19">src/utils/google.ts:19</a></li>
						</ul>
					</aside>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Functions</h2>
				<section class="tsd-panel tsd-member tsd-kind-function tsd-parent-kind-namespace">
					<a name="getfilebackup" class="tsd-anchor"></a>
					<h3>get<wbr>File<wbr>Backup</h3>
					<ul class="tsd-signatures tsd-kind-function tsd-parent-kind-namespace">
						<li class="tsd-signature tsd-kind-icon">get<wbr>File<wbr>Backup<span class="tsd-signature-symbol">(</span>id<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">string</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><a href="../globals.html#filebackup" class="tsd-signature-type">FileBackup</a><span class="tsd-signature-symbol">&gt;</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/fractapp/fractapp/blob/77ffacb/src/utils/google.ts#L104">src/utils/google.ts:104</a></li>
								</ul>
							</aside>
							<h4 class="tsd-parameters-title">Parameters</h4>
							<ul class="tsd-parameters">
								<li>
									<h5>id: <span class="tsd-signature-type">string</span></h5>
								</li>
							</ul>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><a href="../globals.html#filebackup" class="tsd-signature-type">FileBackup</a><span class="tsd-signature-symbol">&gt;</span></h4>
						</li>
					</ul>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-function tsd-parent-kind-namespace">
					<a name="getitems" class="tsd-anchor"></a>
					<h3>get<wbr>Items</h3>
					<ul class="tsd-signatures tsd-kind-function tsd-parent-kind-namespace">
						<li class="tsd-signature tsd-kind-icon">get<wbr>Items<span class="tsd-signature-symbol">(</span>path<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">string</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">Array</span><span class="tsd-signature-symbol">&lt;</span><a href="../globals.html#driveitem" class="tsd-signature-type">DriveItem</a><span class="tsd-signature-symbol">&gt;</span><span class="tsd-signature-symbol">&gt;</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/fractapp/fractapp/blob/77ffacb/src/utils/google.ts#L74">src/utils/google.ts:74</a></li>
								</ul>
							</aside>
							<h4 class="tsd-parameters-title">Parameters</h4>
							<ul class="tsd-parameters">
								<li>
									<h5>path: <span class="tsd-signature-type">string</span></h5>
								</li>
							</ul>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">Array</span><span class="tsd-signature-symbol">&lt;</span><a href="../globals.html#driveitem" class="tsd-signature-type">DriveItem</a><span class="tsd-signature-symbol">&gt;</span><span class="tsd-signature-symbol">&gt;</span></h4>
						</li>
					</ul>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-function tsd-parent-kind-namespace">
					<a name="safesave" class="tsd-anchor"></a>
					<h3>safe<wbr>Save</h3>
					<ul class="tsd-signatures tsd-kind-function tsd-parent-kind-namespace">
						<li class="tsd-signature tsd-kind-icon">safe<wbr>Save<span class="tsd-signature-symbol">(</span>dir<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">string</span>, fileName<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">string</span>, file<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">string</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">boolean</span><span class="tsd-signature-symbol">&gt;</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/fractapp/fractapp/blob/77ffacb/src/utils/google.ts#L44">src/utils/google.ts:44</a></li>
								</ul>
							</aside>
							<h4 class="tsd-parameters-title">Parameters</h4>
							<ul class="tsd-parameters">
								<li>
									<h5>dir: <span class="tsd-signature-type">string</span></h5>
								</li>
								<li>
									<h5>fileName: <span class="tsd-signature-type">string</span></h5>
								</li>
								<li>
									<h5>file: <span class="tsd-signature-type">string</span></h5>
								</li>
							</ul>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">boolean</span><span class="tsd-signature-symbol">&gt;</span></h4>
						</li>
					</ul>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-function tsd-parent-kind-namespace">
					<a name="signin" class="tsd-anchor"></a>
					<h3>sign<wbr>In</h3>
					<ul class="tsd-signatures tsd-kind-function tsd-parent-kind-namespace">
						<li class="tsd-signature tsd-kind-icon">sign<wbr>In<span class="tsd-signature-symbol">(</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">void</span><span class="tsd-signature-symbol">&gt;</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/fractapp/fractapp/blob/77ffacb/src/utils/google.ts#L32">src/utils/google.ts:32</a></li>
								</ul>
							</aside>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">void</span><span class="tsd-signature-symbol">&gt;</span></h4>
						</li>
					</ul>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-function tsd-parent-kind-namespace">
					<a name="signout" class="tsd-anchor"></a>
					<h3>sign<wbr>Out</h3>
					<ul class="tsd-signatures tsd-kind-function tsd-parent-kind-namespace">
						<li class="tsd-signature tsd-kind-icon">sign<wbr>Out<span class="tsd-signature-symbol">(</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">void</span><span class="tsd-signature-symbol">&gt;</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/fractapp/fractapp/blob/77ffacb/src/utils/google.ts#L21">src/utils/google.ts:21</a></li>
								</ul>
							</aside>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">void</span><span class="tsd-signature-symbol">&gt;</span></h4>
						</li>
					</ul>
				</section>
			</section>
		</div>
		<div class="col-4 col-menu menu-sticky-wrap menu-highlight">
			<nav class="tsd-navigation primary">
				<ul>
					<li class="globals  ">
						<a href="../globals.html"><em>Globals</em></a>
					</li>
					<li class=" tsd-kind-namespace">
						<a href="accountsstore.html">Accounts<wbr>Store</a>
					</li>
					<li class=" tsd-kind-namespace">
						<a href="backendapi.html">Backend<wbr>Api</a>
					</li>
					<li class=" tsd-kind-namespace">
						<a href="chatsstore.html">Chats<wbr>Store</a>
					</li>
					<li class=" tsd-kind-namespace">
						<a href="db.html">DB</a>
					</li>
					<li class=" tsd-kind-namespace">
						<a href="dialogstore.html">Dialog<wbr>Store</a>
					</li>
					<li class=" tsd-kind-namespace">
						<a href="globalstore.html">Global<wbr>Store</a>
					</li>
					<li class="current tsd-kind-namespace">
						<a href="google.html">Google</a>
					</li>
					<li class=" tsd-kind-namespace">
						<a href="mathutils.html">Math<wbr>Utils</a>
					</li>
					<li class=" tsd-kind-namespace">
						<a href="passcodeutil.html">Passcode<wbr>Util</a>
					</li>
					<li class=" tsd-kind-namespace">
						<a href="pricesstore.html">Prices<wbr>Store</a>
					</li>
					<li class=" tsd-kind-namespace">
						<a href="stringutils.html">String<wbr>Utils</a>
					</li>
					<li class=" tsd-kind-namespace">
						<a href="task.html">Task</a>
					</li>
					<li class=" tsd-kind-namespace">
						<a href="transactionsstore.html">Transactions<wbr>Store</a>
					</li>
				</ul>
			</nav>
			<nav class="tsd-navigation secondary menu-sticky">
				<ul class="before-current">
					<li class=" tsd-kind-variable tsd-parent-kind-namespace">
						<a href="google.html#foldertype" class="tsd-kind-icon">folder<wbr>Type</a>
					</li>
					<li class=" tsd-kind-variable tsd-parent-kind-namespace">
						<a href="google.html#jsontype" class="tsd-kind-icon">json<wbr>Type</a>
					</li>
					<li class=" tsd-kind-function tsd-parent-kind-namespace">
						<a href="google.html#getfilebackup" class="tsd-kind-icon">get<wbr>File<wbr>Backup</a>
					</li>
					<li class=" tsd-kind-function tsd-parent-kind-namespace">
						<a href="google.html#getitems" class="tsd-kind-icon">get<wbr>Items</a>
					</li>
					<li class=" tsd-kind-function tsd-parent-kind-namespace">
						<a href="google.html#safesave" class="tsd-kind-icon">safe<wbr>Save</a>
					</li>
					<li class=" tsd-kind-function tsd-parent-kind-namespace">
						<a href="google.html#signin" class="tsd-kind-icon">sign<wbr>In</a>
					</li>
					<li class=" tsd-kind-function tsd-parent-kind-namespace">
						<a href="google.html#signout" class="tsd-kind-icon">sign<wbr>Out</a>
					</li>
				</ul>
			</nav>
		</div>
	</div>
</div>
<footer class="with-border-bottom">
	<div class="container">
		<h2>Legend</h2>
		<div class="tsd-legend-group">
			<ul class="tsd-legend">
				<li class="tsd-kind-constructor tsd-parent-kind-class"><span class="tsd-kind-icon">Constructor</span></li>
				<li class="tsd-kind-property tsd-parent-kind-class"><span class="tsd-kind-icon">Property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class"><span class="tsd-kind-icon">Method</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-private"><span class="tsd-kind-icon">Private property</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-method tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static method</span></li>
			</ul>
		</div>
	</div>
</footer>
<div class="container tsd-generator">
	<p>Generated using <a href="https://typedoc.org/" target="_blank">TypeDoc</a></p>
</div>
<div class="overlay"></div>
<script src="../assets/js/main.js"></script>
</body>
</html>