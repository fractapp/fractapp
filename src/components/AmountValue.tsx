import React from 'react';
import {
  StyleSheet,
  Text,
  TextInput,
  TouchableOpacity,
  View,
} from 'react-native';
import {Currency, getSymbol} from 'types/wallet';

/**
 * Text input for amount
 * @category Components
 */
export const AmountValue = ({
  value,
  alternativeValue,
  fee,
  currency,
  isUSDMode,
  width = '100%',
  onPress,
}: {
  value: number;
  alternativeValue: number;
  fee: number;
  currency: Currency;
  isUSDMode: boolean;
  width?: string;
  onPress?: () => void;
}) => {
  return (
    <TouchableOpacity
      style={{width: width, alignItems: 'center'}}
      onPress={onPress}>
      <View
        style={{
          flexDirection: 'row',
          alignItems: 'center',
        }}>
        <Text style={styles.value}>{isUSDMode && value !== 0 && '$'}</Text>
        <TextInput
          style={[
            styles.value,
            {
              color: value === 0 ? '#BFBDBD' : 'black',
              width: value === 0 ? width : 'auto',
              alignSelf: value === 0 ? 'flex-start' : 'center',
              paddingLeft: 0,
              paddingRight: 0,
              paddingBottom: 0,
              paddingTop: 0,
              marginTop: 8,
              marginBottom: 8,
            },
          ]}
          editable={false}
          value={value === 0 ? 'Enter amount' : String(value)}
          keyboardType={'decimal-pad'}
        />
        <Text style={styles.valueCurrency}>
          {!isUSDMode && value !== 0 && ' ' + getSymbol(currency)}
        </Text>
      </View>

      <View style={[styles.line, {width: width}]} />
      <View style={{flexDirection: 'row', width: width}}>
        <View style={{width: '50%', alignItems: 'flex-start'}}>
          {fee !== 0 && <Text style={[styles.subValue]}>Fee ${fee}</Text>}
        </View>
        <View style={{width: '50%', alignItems: 'flex-end'}}>
          {alternativeValue !== 0 && (
            <Text style={styles.subValue}>
              {isUSDMode
                ? `${alternativeValue} ${getSymbol(currency)}`
                : `$${alternativeValue}`}
            </Text>
          )}
        </View>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  line: {
    borderBottomWidth: 1,
    borderColor: '#DADADA',
  },
  value: {
    fontSize: 17,
    fontFamily: 'Roboto-Regular',
    fontStyle: 'normal',
    fontWeight: 'normal',
    color: 'black',
  },
  valueCurrency: {
    fontSize: 17,
    fontFamily: 'Roboto-Regular',
    fontStyle: 'normal',
    fontWeight: 'normal',
    color: '#888888',
  },
  subValue: {
    marginTop: 7,
    fontSize: 15,
    fontFamily: 'Roboto-Regular',
    fontStyle: 'normal',
    fontWeight: 'normal',
    color: 'black',
  },
});
