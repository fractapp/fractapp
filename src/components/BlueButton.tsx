import React from 'react';
import { StyleSheet, Text, View, TouchableOpacity, Image } from 'react-native';


export const BlueButton = ({ text, height, width = "100%", onPress }: { text: string, height: number, width?: string, onPress: () => void} ) => {
    return (
        <TouchableOpacity style={[styles.button, {height: height, width: width} ]} onPress={onPress}>
            <Text style={styles.buttonText}>{text}</Text>
        </TouchableOpacity>
    );
}


const styles = StyleSheet.create({
    buttonText: {
        fontSize: 15,
        fontFamily: "Roboto-Regular",
        fontStyle: "normal",
        fontWeight: "normal",
        color: "white"
    },
    button: {
        backgroundColor: "#2AB2E2",
        borderRadius: 10,
        alignItems: 'center',
        justifyContent: 'center'
    }
});