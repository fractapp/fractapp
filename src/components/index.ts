export {BlueButton} from './BlueButton';
export {WhiteButton, Img} from './WhiteButton';
export {SeedButton} from './SeedButton';
export {StatisticsBar} from './StatisticsBar';
export {PasswordInput} from './PasswordInput';
export {TextInput} from './TextInput';
export {TransactionInfo} from './TransactionInfo';
export {WalletInfo} from './WalletInfo';
export {WalletDetailsInfo} from './WalletDetailsInfo';
export {Dialog} from './Dialog';
export {Loader} from './Loader';
export {DriveItemView} from './DriveItemView';
export {PassCode} from './PassCode';
export {SuccessButton} from './SuccessButton';
export {WalletLogo} from './WalletLogo';
export {TabMenu} from './TabMenu';
export {ChatShortInfo} from './ChatShortInfo';
export {PaymentMsg} from './PaymentMsg';
export {Contact} from './Contact';
export {Receiver} from './Receiver';
export {AmountInput} from './AmountInput';
export {AmountValue} from './AmountValue';
export {ReceiverWithEnterAddress} from './ReceiverWithEnterAddress';
