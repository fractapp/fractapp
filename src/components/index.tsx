export {BlueButton} from './BlueButton';
export {WhiteButton, Img} from './WhiteButton';
export {SeedButton} from './SeedButton';
export {StatisticsBar} from './StatisticsBar';
export {PasswordInput} from './PasswordInput';
export {TextInput} from './TextInput';
export {TransactionInfo} from './TransactionInfo';
export {WalletInfo} from './WalletInfo';
export {WalletDetailsInfo} from './WalletDetailsInfo';
export {Dialog} from './Dialog';
export {Loader} from './Loader';
export {DriveItemView} from './DriveItemView';
export {PassCode} from './PassCode';
export {SuccessButton} from './SuccessButton';
export {WalletLogo} from './WalletLogo';
