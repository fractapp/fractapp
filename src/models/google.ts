/**
 * Back item id for drive item view
 * @category Models
 */
export const BackItemId = 'back';

/**
 * Type Item for drive item view
 * @category Models
 */
export enum Type {
  Dir,
  Json,
}

/**
 * Drive item (foler/file) for google drive
 * @category Models
 */
export type DriveItem = {
  id: string;
  title: string;
  type: Type;
};
