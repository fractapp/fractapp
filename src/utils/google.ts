import { GoogleSignin } from '@react-native-community/google-signin';
import GDrive from "react-native-google-drive-api-wrapper";
import { Type, DiskItem } from 'models/google'
import { FileBackup } from 'models/backup';

const folderType = "application/vnd.google-apps.folder"
const jsonType = "application/json"

GoogleSignin.configure({
    scopes: ['https://www.googleapis.com/auth/drive'],
    offlineAccess: false,
    forceCodeForRefreshToken: true,
})

export async function signOut() {
    const isSigned = await GoogleSignin.isSignedIn()
    if (!isSigned)
        return

    const tokens = await GoogleSignin.getTokens()
    await GoogleSignin.clearCachedAccessToken(tokens.accessToken)
    await GoogleSignin.signOut()
}

export async function signIn() {
    const user = await GoogleSignin.getCurrentUser()
    const isSigned = await GoogleSignin.isSignedIn()
    if (user == null || !isSigned)
        await GoogleSignin.signIn()

    const tokens = await GoogleSignin.getTokens()
    await GDrive.setAccessToken(tokens.accessToken);
    await GDrive.init()
}

export async function safeSave(dir: string, fileName: string, file: string): Promise<boolean> {
    await GDrive.files.safeCreateFolder({
        name: dir,
        parents: ["root"]
    });

    const fieldIdDir = await GDrive.files.getId(dir, ["root"])
    const fieldId = await GDrive.files.getId(fileName, ["root", fieldIdDir])
    if (fieldId == undefined) {
        const result = await GDrive.files.createFileMultipart(
            file,
            "application/json",
            {
                parents: [fieldIdDir],
                name: fileName
            }, false);
        if (result.status != 200)
            return false
    }

    return true
}

export async function getItems(path: string): Promise<Array<DiskItem>> {
    const rs = await GDrive.files.list({ q: `"${path}" in parents and (mimeType = "${folderType}" or mimeType = "${jsonType}")` })
    const rsBody = await rs.json()
    const result = new Array<DiskItem>()
    for (let index in rsBody.files) {
        const file = rsBody.files[index]

        let itemType: Type
        switch (file.mimeType) {
            case folderType:
                itemType = Type.Dir
                break
            case jsonType:
                itemType = Type.Json
                break
            default:
                continue
        }

        result.push(new DiskItem(file.id, file.name, itemType))
    }
    return result
}

export async function getFileBackup(id: string): Promise<FileBackup> {
    const rs = await GDrive.files.get(id, { alt: "media" })
    const rsBody = await rs.json()
    if (rsBody.algorithm == undefined || rsBody.seed == undefined) {
        throw ("Invalid file")
    }
    return rsBody
}