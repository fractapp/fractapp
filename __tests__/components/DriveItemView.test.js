import React from 'react';
import renderer from 'react-test-renderer';
import {DriveItemView} from 'components/DriveItemView';
import {DriveItem, Type} from 'types/google';
import StringUtils from 'utils/string';

jest.mock('react-native-i18n', () => ({
  t: (value) => value,
}));

it('Test id=1 && title="Title #1" type=Dir', () => {
  const tree = renderer
    .create(
      <DriveItemView
        item={{
          id: '1',
          title: 'Title #1',
          type: Type.Dir,
        }}
        onPress={() => console.log('test')}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test id=2 && title="Title #2" type=Json', () => {
  const tree = renderer
    .create(
      <DriveItemView
        item={{
          id: '2',
          title: 'Title #2',
          type: Type.Json,
        }}
        onPress={() => console.log('test')}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test title length > 25', () => {
  const tree = renderer
    .create(
      <DriveItemView
        item={{
          id: '1',
          title: '1exaAg2VJRQbyUBAeXcktChCAqjVP9TUxF3zo23R2T6EGdE',
          type: Type.Dir,
        }}
        onPress={() => console.log('test')}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
