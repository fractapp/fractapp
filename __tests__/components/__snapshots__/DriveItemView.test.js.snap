// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test id=1 && title="Title #1" type=Dir 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#888888",
            "fontSize": 30,
          },
          undefined,
          Object {
            "fontFamily": "Feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "color": "black",
          "fontFamily": "Roboto-Regular",
          "fontSize": 20,
          "marginLeft": 10,
        }
      }
    >
      Title #1
    </Text>
  </View>
</View>
`;

exports[`Test id=2 && title="Title #2" type=Json 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#888888",
            "fontSize": 30,
          },
          undefined,
          Object {
            "fontFamily": "anticon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "color": "black",
          "fontFamily": "Roboto-Regular",
          "fontSize": 20,
          "marginLeft": 10,
        }
      }
    >
      Title #2
    </Text>
  </View>
</View>
`;

exports[`Test title length > 25 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#888888",
            "fontSize": 30,
          },
          undefined,
          Object {
            "fontFamily": "Feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "color": "black",
          "fontFamily": "Roboto-Regular",
          "fontSize": 20,
          "marginLeft": 10,
        }
      }
    >
      1exaAg2VJRQbyUBAeXcktChCA...
    </Text>
  </View>
</View>
`;
